galley/pkg/runtime/projections/serviceentry/convert/convert_bench_test.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package convert_test
+
+import (
+	"testing"
+	"time"
+
+	networking "istio.io/api/networking/v1alpha3"
+	"istio.io/istio/galley/pkg/runtime/projections/serviceentry/convert"
+	"istio.io/istio/galley/pkg/runtime/projections/serviceentry/node"
+	"istio.io/istio/galley/pkg/runtime/projections/serviceentry/pod"
+	"istio.io/istio/galley/pkg/runtime/resource"
+
+	coreV1 "k8s.io/api/core/v1"
+)
+
+const (
+	benchNamespace = "benchmarkns"
+)
+
+func BenchmarkService(b *testing.B) {
+	benchmarkService(b, false)
+}
+
+func benchmarkService(b *testing.B, reuse bool) {
+	b.Helper()
+
+	b.StopTimer()
+
+	// Setup a fairy representative k8s Service ... cluster IP, a few ports, some labels/annotations.
+	fullName := resource.FullNameFromNamespaceAndName(benchNamespace, "someservice")
+	spec := &coreV1.ServiceSpec{
+		ClusterIP: "10.0.0.1",
+		Ports: []coreV1.ServicePort{
+			{
+				Name:     "http",
+				Port:     80,
+				Protocol: coreV1.ProtocolTCP,
+			},
+			{
+				Name:     "https",
+				Port:     443,
+				Protocol: coreV1.ProtocolTCP,
+			},
+			{
+				Name:     "grpc",
+				Port:     8088,
+				Protocol: coreV1.ProtocolTCP,
+			},
+		},
+	}
+	metadata := resource.Metadata{
+		CreateTime: time.Now(),
+		Annotations: resource.Annotations{
+			"Annotation1": "AnnotationValue1",
+			"Annotation2": "AnnotationValue2",
+			"Annotation3": "AnnotationValue3",
+			"Annotation4": "AnnotationValue4",
+			"Annotation5": "AnnotationValue5",
+		},
+		Labels: resource.Labels{
+			"Label1": "LabelValue1",
+			"Label2": "LabelValue2",
+			"Label3": "LabelValue3",
+			"Label4": "LabelValue4",
+			"Label5": "LabelValue5",
+		},
+	}
+
+	// Create/init the output ServiceEntry if reuse is enabled.
+	var out *networking.ServiceEntry
+	if reuse {
+		out = &networking.ServiceEntry{}
+		convertService(spec, metadata, fullName, out)
+	}
+
+	b.StartTimer()
+
+	for i := 0; i < b.N; i++ {
```

Comment:
Minor nit: if you use t.Run(...) equivalent of b here, then you don't need to reset timer. It should also account memory allocations only within the context of the lambda.